src/libraries/System.Private.CoreLib/src/System/Threading/AutoreleasePool.cs

```diff
@@ -8,21 +8,9 @@ namespace System.Threading
 {
     internal static class AutoreleasePool
     {
-        private static bool CheckEnableAutoreleasePool()
-        {
-            const string feature = "System.Threading.Thread.EnableAutoreleasePool";
-#if !CORECLR
-            return AppContextConfigHelper.GetBooleanConfig(feature, false);
-#else
-            bool isEnabled = CLRConfig.GetBoolValue(feature, out bool isSet);
-            if (!isSet)
-                return false;
-
-            return isEnabled;
-#endif
-        }
-
-        public static bool EnableAutoreleasePool { get; } = CheckEnableAutoreleasePool();
+        private static bool? _isAutoreleasePoolEnabled;
```

Comment:
This is not thread safe. Assignment of `bool?` is not atomic.

Reply:
One of the usual ways to solve this is to use 0/1/-1 like https://github.com/dotnet/runtime/blob/55cea645dc52a6f7d5e5105846ec49bc7fda2a40/src/libraries/System.Private.CoreLib/src/System/LocalAppContextSwitches.cs#L54

Reply:
Nice! much better than what I was testing 😄
```diff
+        private static readonly object s_syncRoot = new object();
         private static bool? _isAutoreleasePoolEnabled;
-        public static bool EnableAutoreleasePool => (_isAutoreleasePoolEnabled ??
-            (_isAutoreleasePoolEnabled = AppContextConfigHelper.GetBooleanConfig("System.Threading.Thread.EnableAutoreleasePool", false))).Value;
+        public static bool EnableAutoreleasePool
+        {
+            get
+            {
+                if (_isAutoreleasePoolEnabled.HasValue) return _isAutoreleasePoolEnabled.Value;
+
+                lock (s_syncRoot)
+                {
+                    if (_isAutoreleasePoolEnabled.HasValue) return _isAutoreleasePoolEnabled.Value;
+
+                    _isAutoreleasePoolEnabled = AppContextConfigHelper.GetBooleanConfig("System.Threading.Thread.EnableAutoreleasePool", false);
+                }
+
+                return _isAutoreleasePoolEnabled.Value;
+            }
+        }
```